nextcord/client.py

```diff
@@ -574,10 +574,12 @@ async def on_application_command_error(
         if interaction.application_command and interaction.application_command.has_error_handler():
             return
 
-        # TODO implement cog error handling
-        # cog = context.cog
-        # if cog and cog.has_error_handler():
-        #     return
+        if interaction.application_command is None:
+            return  # Not supposed to ever happen
+
+        cog = interaction.application_command.parent_cog
+        if cog and cog.has_application_command_error_handler():
+            return
```

Comment:
Sound logic, but wouldn't this be better suited to living in `application_command.has_error_handler`?

Reply:
i simply copied the structure from the prefix commands and all that. Thats how it works there, so i think we should keep it consistend and keep it with slash commands like this as well.